internal/controller/promotions/promotions.go

```diff
@@ -665,3 +661,30 @@ func parseCreateActorAnnotation(promo *kargoapi.Promotion) string {
 	}
 	return creator
 }
+
+func calculateRequeueInterval(p *kargoapi.Promotion) time.Duration {
+	defaultRequeueInterval := 5 * time.Minute
+	for i := range p.Spec.Steps {
```

Comment:
Why iterate here when we know which element we're looking for and could grab it directly without looking for it?